src/unix/netbsd.c

```diff
@@ -79,19 +74,22 @@ int uv_exepath(char* buffer, size_t* size) {
   mib[1] = KERN_PROC_ARGS;
   mib[2] = -1;
   mib[3] = KERN_PROC_PATHNAME;
-  int_size = ARRAY_SIZE(int_buf);
-
-  if (sysctl(mib, 4, int_buf, &int_size, NULL, 0))
-    return UV__ERR(errno);
-
-  /* Copy string from the intermediate buffer to outer one with appropriate
-   * length.
-   */
-  /* TODO(bnoordhuis) Check uv__strscpy() return value. */
-  uv__strscpy(buffer, int_buf, *size);
+  saved_errno = errno;
+  errno = 0;
+
+  if (sysctl(mib, ARRAY_SIZE(mib), buffer, size, NULL, 0)) {
+    if (errno != ENOSPC) {
+      int err = UV__ERR(errno);
+      errno = saved_errno;
+      return err;
+    }
+  }
 
-  /* Set new size. */
-  *size = strlen(buffer);
+  /* In case the buffer might not be C-string complete, adding trailing end */
+  if (errno == 0 || errno == ENOSPC)
+    (*size)--;
+  buffer[*size] = '\0';
+  errno = saved_errno;
```

Comment:
```suggestion
```
libuv doesn't usually preserve errno (though I suppose it couldn't hurt) but instead keeps it in a local variable for places it needs it